### PR TITLE
Feature: GQL Context (#58635)

### PR DESCRIPTION
**Pull request by nx-mukesh**
Ticket:-
https://redmine.successive.tech/issues/58635

Description:-
Context is an object shared by all the resolvers of a specific execution. It's useful for keeping data such as authentication info, the current user, database connection, data sources and other things we need for running your business logic.

### CODE REVIEW

File: apollo-training/src/modules/trainee/query.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
 import user from '../../service/user';
 
 export default {
-  getAllTrainees: () => user.getAllUsers(),
+  getAllTrainees: async (parent, args, context) => {
+    console.log(args);
```
Comment (permil-successive) [inline apollo-training/src/modules/trainee/query.js]: Please remove this console

File: apollo-training/src/modules/trainee/query.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+    const response = await context.dataSources.traineeAPI.getTrainee(args);
+    // const result1 = response.data.map((element) => (
+    //   { name: element[0].name, email: element[0].email }
+    // ));
+    console.log('response.data.result --------------> ', response.data[0].result);
```
Comment (permil-successive) [inline apollo-training/src/modules/trainee/query.js]: please remove

File: apollo-training/src/server.js [hunk 3]
```diff
@@ -56,7 +56,11 @@ export default class Server {
         userAPI: new UserAPI(),
         traineeAPI: new TraineeAPI(),
       }),
-
+      context: ({ req }) => {
+        const token = req.headers.authorization || '';
+        console.log('token at gql: ', token);
```
Comment (permil-successive) [inline apollo-training/src/server.js]: please remove this console

File: apollo-training/src/modules/trainee/query.js (hunk 2 above)
Comment (nx-mukesh) [reply apollo-training/src/modules/trainee/query.js]: removed

File: apollo-training/src/server.js (hunk 3 above)
Comment (nx-mukesh) [reply apollo-training/src/server.js]: fixed

**Review by permil-successive — APPROVED** [review]
Approved

**Pull request merged**